### PR TITLE
Join link

### PR DESCRIPTION
**Pull request by aroquev00**
LAST PRRRRR
Now when the teacher creates a gameInstance there is a link the teacher can share with the students for them to join. If the teacher clicks the link it gets copied to the clipboard so that its ready to be shared with the students.

I moved the process so that when the student introduces the room code in the join page, it only redirects to the play screen, and now that is where the POST request is sent to the back end for the student to be registered and assigned an animal alias. Actually this is better because we now have one query instead of two as previously.

image

### CODE REVIEW

File: src/main/java/com/google/sps/daos/DatabaseUserDao.java [hunk 1]
```diff
@@ -117,7 +118,7 @@ public void createIfNotExists(String idToken) {
   }
 
   @Override
-  public void joinGameInstance(String idToken, String gameInstanceId) {
+  public String joinGameInstance(String idToken, String gameInstanceId) {
```
Comment (seanjohnite) [inline src/main/java/com/google/sps/daos/DatabaseUserDao.java:121]: Document this return value.

File: src/main/webapp/teacher/controlGameInstance/controlGameInstance.js [hunk 2]
```diff
@@ -148,6 +148,12 @@ function buildActiveGameInstanceUI({ gameInstanceId, gameInstance, game} = {}) {
 function addGameInstanceIdToUI(gameInstanceId) {
   const gameInstanceIdElement = document.getElementById("jsGameInstanceId");
   gameInstanceIdElement.innerText = "This gameInstance's ID is: " + gameInstanceId;
+
+  const shareGameInstanceElement = document.getElementById('jsShareGameInstance');
+  shareGameInstanceElement.innerText = "Share this link with your students for them to join (click to copy to clipboard): " + "https://quizzy-step-2020.uc.r.appspot.com/student/play-game.html?gameInstanceId=" + gameInstanceId;
```
Comment (seanjohnite) [inline src/main/webapp/teacher/controlGameInstance/controlGameInstance.js:153]: does this element have have cursor: pointer set on CSS?

File: src/main/webapp/teacher/controlGameInstance/controlGameInstance.js [hunk 3]
```diff
@@ -479,9 +485,7 @@ function addQuestionAnswerToHistoryUI({ questionId, answerId, answer } = {}) {
   answerInQuestionStatsDivElement.innerText = answer.title + ' with ' + answer.numberAnswers + ' answers.';
 
   if (answer.correct) {
-    questionAnswerDivElement.innerText = 'Correctly answered, chose: "' + answer.chosen + '"';
-  } else {
-    questionAnswerDivElement.innerText = 'Incorrect answer, chose: "' + answer.chosen + '"';
```
Comment (seanjohnite) [inline src/main/webapp/teacher/controlGameInstance/controlGameInstance.js:484]: removed this on purpose?

**Review by seanjohnite — APPROVED** [review]
LG!

File: src/main/webapp/teacher/controlGameInstance/controlGameInstance.js (hunk 2 above)
Comment (aroquev00) [reply src/main/webapp/teacher/controlGameInstance/controlGameInstance.js:153]: Yep it does!

File: src/main/webapp/teacher/controlGameInstance/controlGameInstance.js (hunk 3 above)
Comment (aroquev00) [reply src/main/webapp/teacher/controlGameInstance/controlGameInstance.js:484]: Yep that was there by mistake, was accessing nonexistent info and changed it to display the actual answer info.

File: src/main/java/com/google/sps/daos/DatabaseUserDao.java (hunk 1 above)
Comment (aroquev00) [reply src/main/java/com/google/sps/daos/DatabaseUserDao.java:123]: Done! Let me know if its missing more explanation.

**Pull request merged**